backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/SubscriptionsController.java

```diff
@@ -98,6 +102,44 @@ public SubscriptionModel getSubscriptionById(
 		return subscriptionModelMapper.toModel(userId, subscription);
 	}
 
+	@PutMapping({ "/{subscriptionId}" })
+	@Operation(summary = "Update a subscriptions by ID")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void updateSubscriptionById(
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The id of the user.", example = "00000000-0000-0000-0000-000000000000")
+			@PathVariable String userId,
+
+			@NotBlank(message = "subscriptionId must not be null or blank")
+			@Parameter(description = "The id of the subscription.", example = "00000000-0000-0000-0000-000000000000")
+			@PathVariable String subscriptionId,
+			
+			@Validated @RequestBody PreferredLanguageModel preferredLanguageModel) {
+		
+		userService.updateSubscriptionForUser(userId, subscriptionId, preferredLanguageModel);
```

Comment:
You should:
- check that the user exists and throw a 404 (ResourceNotFoundException) if not
- check that the subscription exists and throw a 404 if not

Reply:
fixed